### DISCUSSION TITLE
Is maxWidth really needed on Popup

### DISCUSSION THREAD

**Discussion by GaryHeaven** [Leaflet/Leaflet#8441] [Q&A]
The popup has limits for minimum and maximum width. The default for these (5px and 300px) are set. The 300px invariably is too small these days (500px better) but I am wondering why there is a width limitation at all in leaflet. Do bad things happen when the popup is much bigger than the view window?

**Comment by IvanSanchez** [comment]
Doing a fair amount of `git blame`ing, turns out that the `maxWidth` of `L.Popup` was introduced in https://github.com/Leaflet/Leaflet/commit/739c8e289ec2ce63e430e44e4a89e6db4a8b8155 , which is dated december 2010. Which means Chrome v7, Firefox v3.6 and MSIE v8 were the bleeding edge (with support for MSIE7 and possible MSIE6 too).

Back then, CSS positioning in web browsers was.... *less than optimal*. Specially if I'm right about Leaflet having MSIE6 support, which back then had a 5% market share still.

So the maxwidth is there to manually set the width of the popup, according to this little gem of legacy code:

https://github.com/Leaflet/Leaflet/blob/739c8e289ec2ce63e430e44e4a89e6db4a8b8155/src/layer/Popup.js#L69-L71

So, yes, this is one of those things that should go away when Leaflet finally deprecates MSIE once and forever in #6136 . Or at least move it to a CSS rule where it should be much cleaner.

**Reply by jonkoops** [reply]
Seems like the time has arrived for this to be refactored.

**Comment by Malvoz** [comment]
> Do bad things happen when the popup is much bigger than the view window?

That'd be somewhat of an accessibility issue, similar to https://github.com/Leaflet/Leaflet/issues/2324.

Related issue: https://github.com/Leaflet/Leaflet/issues/8645.
